dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java

```diff
@@ -109,6 +109,74 @@ public final Response createContentPortlet(@Context final HttpServletRequest req
     return response;
   }
 
+    @PUT
+    @Path("/custom/{portletId}/_addtolayout/{layoutId}")
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response addContentPortletToLayout(@Context final HttpServletRequest request,
+            @PathParam("portletId") final String portletId,
+            @PathParam("layoutId") final String layoutId) {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, null)
+                .rejectWhenNoUser(true)
+                .requiredPortlet("roles")
+                .init();
+
+        final User user = initData.getUser();
+
+        Response response = null;
+
+        try {
+
+            final PortletAPI portletAPI = APILocator.getPortletAPI();
+            final LayoutAPI layoutAPI = APILocator.getLayoutAPI();
+
+            if(!portletAPI.canAddPortletToLayout(portletId)){
+                return ResponseUtil.INSTANCE.getErrorResponse(request, Response.Status.UNAUTHORIZED, user.getLocale(),
+                        user.getUserId(),
+                        String.format("Portlet with id %s is restricted and can not be added to a layout.",portletId));
+            }
+
+            final Portlet portlet = portletAPI.findPortlet(portletId);
+            if (null == portlet) {
+                throw new DoesNotExistException(
+                        String.format("Portlet id by %s wasn't found.", portletId));
+            }
+
+            final Layout layout = layoutAPI.loadLayout(layoutId);
+            if (null == layout) {
+                throw new DoesNotExistException(
+                        String.format("Layout id by %s wasn't found.", layoutId));
+            }
+
+            final List<Layout> userLayouts = layoutAPI.loadLayoutsForUser(user);
+            if(!userLayouts.contains(layout)){
+                return ResponseUtil.INSTANCE.getErrorResponse(request, Response.Status.UNAUTHORIZED, user.getLocale(),
+                        user.getUserId(),
+                        "Current user does not have access to the given layout.");
+            }
+
+            final List<String> portletIds = new ArrayList<>(layout.getPortletIds());
+            portletIds.add(portlet.getPortletId());
+            layoutAPI.setPortletIdsToLayout(layout, portletIds);
+
+            return Response.ok(new ResponseEntityView(
+                    map("portlet", portlet.getPortletId(), "layout", layout.getId())))
+                    .build();
+
+        } catch (Exception e) {
```

Comment:
you can remove the catch, Jersey handles the exceptions for us

Reply:
done